controllers/goharbor/core/resources.go

```diff
@@ -59,5 +59,17 @@ func (r *Reconciler) AddResources(ctx context.Context, resource resources.Resour
 		return errors.Wrapf(err, "cannot add deployment %s", deployment.GetName())
 	}
 
+	areNetworkPoliciesEnabled, err := r.AreNetworkPoliciesEnabled(ctx, core)
+	if err != nil {
+		return errors.Wrapf(err, "cannot get network policies status")
+	}
+
+	if areNetworkPoliciesEnabled {
+		_, err = r.AddIngressNetworkPolicy(ctx, core)
+		if err != nil {
+			return errors.Wrapf(err, "ingress network policy")
+		}
+	}
+
 	return nil
```

Comment:
Same here, handle edge cases (network policies disabled) at the beginning of the function. It will be more readable.